Sources/Db/Schema/Column.php

```diff
@@ -0,0 +1,203 @@
+	 */
+	public string $type;
+
+	/**
+	 * @var ?int
+	 *
+	 * Size of the column.
+	 * Only applicable to some data types.
+	 */
+	public ?int $size;
+
+	/**
+	 * @var ?bool
+	 *
+	 * Whether the column uses unsigned numerical values.
+	 * Only applicable in MySQL.
+	 */
+	public ?bool $unsigned;
+
+	/**
+	 * @var ?bool
+	 *
+	 * Whether the column disallows null values.
+	 */
+	public ?bool $not_null;
+
+	/**
+	 * @var string|float|int|bool|null
+	 *
+	 * Default value of the column.
+	 */
+	public string|float|int|bool|null $default;
+
+	/**
+	 * @var ?bool
+	 *
+	 * Whether this is an automatically incrementing column.
+	 * Only applicable to integer columns.
+	 */
+	public ?bool $auto;
+
+	/**
+	 * @var ?string
+	 *
+	 * The character set for string data.
+	 * Only applicable to string types.
+	 */
+	public ?string $charset;
+
+	/**
+	 * @var bool
+	 *
+	 * Set this to true to drop the default during an ALTER TABLE operation.
+	 *
+	 * This is not set by __construct(). It can be set afterward.
+	 */
+	public bool $drop_default = false;
+
+	/****************
+	 * Public methods
+	 ****************/
+
+	/**
+	 * Constructor.
+	 *
+	 * @param string $name Name of the column.
+	 * @param string $type Data type of the column.
+	 * @param ?int $size Size of the column.
+	 *    Only applicable to some data types.
+	 * @param ?bool $unsigned Whether the column uses unsigned numerical values.
+	 *    Only used by MySQL.
+	 * @param ?bool $not_null Whether the column disallows null values.
+	 * @param mixed $default Default value of the column. If null, no default
+	 *    will be set. To set an explicit null default, use the string 'NULL'.
+	 * @param ?bool $auto Whether this is an automatically incrementing column.
+	 *    Only applicable to integer columns.
+	 * @param ?string $charset The character set for string data.
+	 *    Only applicable to string types. If null, will be set automatically.
+	 */
+	public function __construct(
+		string $name,
+		string $type,
+		?int $size = null,
+		?bool $unsigned = null,
+		?bool $not_null = null,
+		string|float|int|bool|null $default = null,
+		?bool $auto = null,
+		?string $charset = null,
+	) {
+		$this->name = strtolower($name);
+		$this->type = strtolower($type);
+
+		if (isset($default)) {
+			$this->default = $default === 'NULL' ? null : $default;
+		}
+
+		foreach (['auto', 'size', 'unsigned', 'not_null'] as $var) {
+			if (isset($var)) {
+				$this->{$var} = ${$var};
+			}
+		}
+
+		if (isset($charset)) {
+			$this->charset = strtolower($charset);
+		}
+	}
+
+	/**
+	 * Adds this column to the specified table.
+	 *
+	 * @see SMF\Db\DatabaseApi::add_column
+	 *
+	 * @param string $table_name The name of the table to add the column to.
+	 * @param string $if_exists What to do if the column exists.
+	 *    If 'update', column is updated.
+	 * @return bool Whether or not the operation was successful.
+	 */
+	public function add(string $table_name, string $if_exists = 'update'): bool
```

Comment:
This method is the one I reference in my other comment. 

Why should columns be concerned with tables? Tables should be concerned about columns. Not vice-versa.

The Table class should expose an addColumn method. The Table class already knows which db table its tied to by its $name property. Well, I say it should expose an addColumn. Honestly there should be a group of DDL objects that expose this, and Column should be the parent.

Question:
How, in the future, do you propose to test that the column this instance represents is representative of a Binary, DateTime or TimeStamp for example? We cant test based on type as that is apparent because we do not have objects that represent those "types". So the only choice is to test the instances state since this Column instance can possibly represent all "types". It handles all responsibilities. It should handles only the common responsibilities shared for all column types. Int and text are very different creatures. SRP in OO is a thing for a reason.

The behavioral contract here is that tables contain columns. You do not write queries of the nature sudo code:
COULMN ALTER TABLE 
its ALTER TABLE ADD, DROP, MODIFY
You can not get there from a column. Any code / API modeling it should not be able to get there either. Its a breach of contract in respect to the objects responsibility. What it represents in the context of the system should not be able to perform the action this method provides.